pennylane/debugging.py

```diff
@@ -53,21 +53,61 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         self.device._debugger = None
 
 
-def snapshots(qnode):
-    r"""Create a function that retrieves snapshot results from a QNode.
+@transform
+def snapshots(tape: QuantumTape) -> Tuple[Sequence[QuantumTape], Callable]:
+    r"""Transforms a QNode into several tapes by aggregating all operations up to a `qml.Snapshot`
+    operation into their own execution tape.
+
+    The transform is conservative about the wires that it includes in each tape.
+    So, if all operations preceding a snapshot in a 3-qubit circuit has been applied to only one wire,
+    the tape would only be looking at this wire. This can be overriden by the configuration of the execution device
+    and its nature. For supported simulators, measurements with no specified wires would default to all wires.
+
+    The output is a dictionary where each key is either the tag supplied to the snapshot or its
+    index in order of appearance, in additition to an "execution_results" that returns the final output
+    of the quantum circuit. The post-processing function is responsible for aggregating the results into
+    this dictionary.
+
+    When the transform is applied to a QNode, the `shots` configuration would be inherited by the `Snapshot`
+    instances, as long as the chosen measurement supports the provided value and is permitted by the device.
+
+    It should be noted that for unsupported devices, a separate execution would be carried out for each snapshot.
+    As illustrated in one of the examples below.
```

Comment:
Great information! 

Let's move this text down below the `Args:`/ `Returns:` section.  We try and keep what's above the args/ returns section brief, but then fill out more details below.